src/shogun/util/factory.h

```diff
@@ -337,5 +334,121 @@ namespace shogun
 	{
 		return std::make_shared<PipelineBuilder>();
 	}
+	} // namespace details
+	// fix me
+	//#ifdef SWIG
+	template <typename TypeName, typename... Args>
+	std::shared_ptr<TypeName> create(Args... args)
+	//#else
+	// template <typename TypeName, typename... Args>
+	// std::shared_ptr<TypeName> create(Args&&... args)
+	//#endif
+	{
+		if constexpr (std::is_same_v<TypeName, Features>)
+		{
+			return details::features(std::forward<Args>(args)...);
+		}
+		else if constexpr (std::is_same_v<TypeName, Labels>)
+		{
+			return details::labels(std::forward<Args>(args)...);
+		}
+		else if constexpr (std::is_same_v<TypeName, CSVFile>)
+		{
+			return details::csv_file(std::forward<Args>(args)...);
+		}
+		else if constexpr (std::is_same_v<TypeName, LibSVMFile>)
+		{
+			return details::libsvm_file(std::forward<Args>(args)...);
+		}
+		else if constexpr (std::is_same_v<TypeName, PipelineBuilder>)
+		{
+			return details::pipeline();
+		}
+		else if constexpr (std::is_same_v<TypeName, Kernel>)
+		{
+			return details::kernel(std::forward<Args>(args)...);
+		}
+		else
+		{
+			static_assert(
+			    (sizeof...(Args) == 1) ||
+			        (std::is_constructible_v<Args, std::string> && ...),
+			    "The create function requires a string argument.");
+			return create_object<TypeName>(
+			    (std::string{std::forward<Args>(args)}.c_str())...);
+		}
+	}
+
+	// The SWIG now only support one template parameter in variadic template, so
+	// we have to write those create_ wrapper function
```

Comment:
could you move these swig specific things to `factory.i` please? You can just write C++ code there using `%{...C++ code...}%` syntax